modules/navier_stokes/src/userobjects/NSFVRhieChowInterpolator.C

```diff
@@ -0,0 +1,582 @@
+    paramError("boundaries_to_force",
+               "Do not use the boundaries_to_force parameter to control execution of INSFV "
+               "advection objects");
+}
+
+void
+NSFVRhieChowInterpolator::initialize()
+{
+  std::set<BoundaryID> all_connected_boundaries;
+  const auto & blk_ids = blockRestricted() ? blockIDs() : _mesh.meshSubdomains();
+  for (const auto blk_id : blk_ids)
+  {
+    const auto & connected_boundaries = _mesh.getSubdomainBoundaryIds(blk_id);
+    for (const auto bnd_id : connected_boundaries)
+      all_connected_boundaries.insert(bnd_id);
+  }
+
+  for (const auto bnd_id : all_connected_boundaries)
+  {
+    setupFlowBoundaries(bnd_id);
+    setupBoundaries<INSFVNoSlipWallBC>(
+        bnd_id, INSFVBCs::INSFVNoSlipWallBC, _no_slip_wall_boundaries);
+    setupBoundaries<INSFVSlipWallBC>(bnd_id, INSFVBCs::INSFVSlipWallBC, _slip_wall_boundaries);
+    setupBoundaries<INSFVSymmetryBC>(bnd_id, INSFVBCs::INSFVSymmetryBC, _symmetry_boundaries);
+  }
+}
+
+void
+NSFVRhieChowInterpolator::setupFlowBoundaries(const BoundaryID bnd_id)
+{
+  std::vector<INSFVFlowBC *> flow_bcs;
+
+  this->_subproblem.getMooseApp()
+      .theWarehouse()
+      .query()
+      .template condition<AttribBoundaries>(bnd_id)
+      .template condition<AttribINSFVBCs>(INSFVBCs::INSFVFlowBC)
+      .queryInto(flow_bcs);
+
+  if (!flow_bcs.empty())
+  {
+    if (dynamic_cast<INSFVFullyDevelopedFlowBC *>(flow_bcs.front()))
+    {
+      _fully_developed_flow_boundaries.insert(bnd_id);
+
+#ifndef NDEBUG
+      for (auto * flow_bc : flow_bcs)
+        mooseAssert(dynamic_cast<INSFVFullyDevelopedFlowBC *>(flow_bc),
+                    "If one BC is a fully developed flow BC, then all other flow BCs on that "
+                    "boundary must also be fully developed flow BCs");
+    }
+    else
+      for (auto * flow_bc : flow_bcs)
+        mooseAssert(!dynamic_cast<INSFVFullyDevelopedFlowBC *>(flow_bc),
+                    "If one BC is not a fully developed flow BC, then all other flow BCs on that "
+                    "boundary must also not be fully developed flow BCs");
+#else
+    }
+#endif
+
+    _flow_boundaries.insert(bnd_id);
+    _all_boundaries.insert(bnd_id);
+  }
+}
+
+template <typename T>
+void
+NSFVRhieChowInterpolator::setupBoundaries(const BoundaryID bnd_id,
+                                          const INSFVBCs bc_type,
+                                          std::set<BoundaryID> & bnd_ids)
+{
+  std::vector<T *> bcs;
+
+  this->_subproblem.getMooseApp()
+      .theWarehouse()
+      .query()
+      .template condition<AttribBoundaries>(bnd_id)
+      .template condition<AttribINSFVBCs>(bc_type)
+      .queryInto(bcs);
+
+  if (!bcs.empty())
+  {
+    bnd_ids.insert(bnd_id);
+    _all_boundaries.insert(bnd_id);
+  }
+}
+
+void
+NSFVRhieChowInterpolator::execute()
+{
+  // Compute all Rhie Chow coefficients, see if it brings better performance
+  // Could be another memory saving feature
```

Comment:
How would this save cpu/memory?

Reply:
If you don't cache RC coefficients in a map, you save another n_faces sized map 
Agree this comment seems to say caching it would save memory which isn't true

Reply:
It seems like you're going to have to save something no matter what. If you execute this during a user object loop you'll have to save off the computations in a data structure on a per face basis. If you do things "on-the-fly" then you'll have to make the choice of whether to do the computation every time information is requested by a consumer or cache and use memory